Publish prereleases (e.g. `v0.15.0-alpha-05`) to NPM, not just GitHub

## Summary

Whenever we make prereleases that include breaking changes, publish them on NPM via the `next` tag (e.g. `npm i purescript@next`).

## Motivation

Before we can make a breaking change release (e.g. `0.15.0`), we have to update at least the core libraries to compile on and and build tools to compile with the upcoming compiler version. So, we made prereleases by publishing them on GitHub but not on NPM (e.g. `v0.14.0-rc1`, `v0.15.0-alpha-05`, etc.). Each project's CI had to be updated to download this GitHub release. 

For libraries, this is no longer a problem. During this `0.15.0` release/migration work, we've switched to using [`setup-purescript`](https://github.com/purescript-contrib/setup-purescript) to get the prerelease. However, this tool only works because it was updated to do that during the `0.15.0` work.

For build tools, this is still a bit of a problem. While we could use `setup-purescript` to download the prerelease, it's more awkward to use. `pulp` and `spago` may be testing their code against multiple versions of the `purs` binary and against multiple OSes. So, using a custom download script or multiple `setup-purescript` steps rather than a build matrix where each build is testing one version is awkward. It would be easier if `npm i -g purescript@next` could be used instead.

Finally, publishing the prerelease on NPM would allow others to use it early and see if there are any bugs that haven't been reported yet. During the `0.15.0` work, a few bugs were found by people testing it out:
- #4274
- https://github.com/purescript-web/purescript-web-html/pull/75

## Proposal

Fixing this would require updating the `RELEASE_GUIDE.md` to further explain that one should publish a pre-release on NPM (besides the other updates we should do for that file). Specifically, the below line in the file...
> After all of the prebuilt binaries are present on the GitHub releases page, publish to npm: change to the npm-package directory and run npm publish. It's a good idea to check that the package can be installed from npm at this point.
should be updated to say something like, "When making a prerelease, change to the `npm-package` directory and run `npm publish --tag next`.

See https://manpages.org/npm-publish

Just published the prerelease on NPM and then tried to install it locally via `npm i purescript@next`. I got this error:
```
$ npm i purescript@next
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm ERR! code 1
npm ERR! path /home/jordan/Programming/Projects/za/node_modules/purescript
npm ERR! command failed
npm ERR! command sh -c install-purescript --purs-ver=0.15.0-alpha-06
npm ERR! [ FAILURE ] Check if a prebuilt 0.15.0-alpha-06 binary is provided for linux
npm ERR! Error: Expected `version` option to be a string of PureScript version, for example '0.12.5', but got an invalid version '0.15.0-alpha-06'.
npm ERR!     at /home/jordan/Programming/Projects/za/node_modules/purescript-installer/download-purescript/index.js:97:20
npm ERR!     at new Subscription (/home/jordan/Programming/Projects/za/node_modules/zen-observable/lib/Observable.js:197:34)
npm ERR!     at Observable.subscribe (/home/jordan/Programming/Projects/za/node_modules/zen-observable/lib/Observable.js:279:14)
npm ERR!     at /home/jordan/Programming/Projects/za/node_modules/purescript-installer/download-or-build-purescript/index.js:266:6
npm ERR!     at new Subscription (/home/jordan/Programming/Projects/za/node_modules/zen-observable/lib/Observable.js:197:34)
npm ERR!     at Observable.subscribe (/home/jordan/Programming/Projects/za/node_modules/zen-observable/lib/Observable.js:279:14)
npm ERR!     at /home/jordan/Programming/Projects/za/node_modules/purescript-installer/install-purescript/index.js:133:62
npm ERR!     at new Subscription (/home/jordan/Programming/Projects/za/node_modules/zen-observable/lib/Observable.js:197:34)
npm ERR!     at Observable.subscribe (/home/jordan/Programming/Projects/za/node_modules/zen-observable/lib/Observable.js:279:14)
npm ERR!     at Object.<anonymous> (/home/jordan/Programming/Projects/za/node_modules/purescript-installer/index.js:370:4)
npm ERR!     at Module._compile (node:internal/modules/cjs/loader:1101:14)
npm ERR!     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
npm ERR!     at Module.load (node:internal/modules/cjs/loader:981:32)
npm ERR!     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
npm ERR!     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
npm ERR!     at node:internal/main/run_main_module:17:47
```

--- a/app/Version.hs
+++ b/app/Version.hs
@@ -17,7 +17,7 @@ import qualified Development.GitRev as GitRev
 -- prerelease identifier here (if any). When releasing a proper version, simply
 -- set this to an empty string.
 prerelease :: String
-prerelease = "-alpha-06"
+prerelease = "-alpha-07"
 
 versionString :: String
 versionString = showVersion Paths.version ++ prerelease ++ extra

--- a/npm-package/package.json
+++ b/npm-package/package.json
@@ -1,6 +1,6 @@
 {
   "name": "purescript",
-  "version": "0.15.0-alpha-06",
+  "version": "0.15.0-alpha-07",
   "license": "ISC",
   "description": "PureScript wrapper that makes it available as a local dependency",
   "author": {
@@ -43,7 +43,7 @@
   ],
   "scripts": {
     "prepublishOnly": "node -e \"require('fs').copyFileSync('purs.bin.placeholder', 'purs.bin');\"",
-    "postinstall": "install-purescript --purs-ver=0.15.0-alpha-06",
+    "postinstall": "install-purescript --purs-ver=0.15.0-alpha-07",
     "test": "echo 'Error: no test specified' && exit 1"
   }
 }

--- a/npm-package/package.json
+++ b/npm-package/package.json
@@ -15,7 +15,7 @@
     "purs": "purs.bin"
   },
   "dependencies": {
-    "purescript-installer": "^0.2.0"
+    "purescript-installer": "^0.2.6"
   },
   "homepage": "https://github.com/purescript/purescript",
   "repository": {

--- a/purescript.cabal
+++ b/purescript.cabal
@@ -2,7 +2,7 @@ cabal-version:  2.4
 
 name:           purescript
 -- note: When updating the prerelease identifier, update it in app/Version.hs too!
-version:        0.15.0-alpha-06
+version:        0.15.0-alpha-07
 synopsis:       PureScript Programming Language Compiler
 description:    A small strongly, statically typed programming language with expressive types, inspired by Haskell and compiling to JavaScript.
 category:       Language

